### PR TITLE
feat(rest): new endpoint merge components

### PR DESCRIPTION
**Pull request by muhammadali9699**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * This PR is to merge two components

Issue: #1957

{
   "type":"component",
   "name":"Deemo comp 4",
   "attachments":[      
   ],
   "createdOn":"2023-05-17",
   "componentType":"INNER_SOURCE",
   "createdBy":"admin@sw360.org",
   "subscribers":[      
   ],
   "moderators":[      
   ],
   "roles":{      
   },
   "visbility":"EVERYONE",
   "externalIds":{      
   },
   "additionalData":{      
   },
   "releases":[      
   ],
   "categories":[
      "virtual-machine",
      "big-data",
      "javaee"
   ]
}

###Screenshot
Screenshot 2023-05-25 235146

### CODE REVIEW

**Comment by eldrin30** [comment]
@muhammadali9699, resolve the conflicts

**Comment by eldrin30** [comment]
Testing was successful

**Comment by heliocastro** [comment]
Please rebase

**Comment by muhammadali9699** [comment]
@heliocastro Rebase successful.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java [hunk 1]
```diff
@@ -695,4 +695,23 @@ private ImportBomRequestPreparation handleImportBomRequestPreparation(ImportBomR
         }
         return importBomRequestPreparationResponse;
     }
+
+    @PreAuthorize("hasAuthority('WRITE')")
+    @RequestMapping(value = COMPONENTS_URL + "/mergecomponents", method = RequestMethod.POST)
```
Comment (ag4ums) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java:700]: PUT seems more appropriate here as the component is getting updated

**Comment by eldrin30** [comment]
Code looks good, kindly rebase

**Comment by muhammadali9699** [comment]
@eldrin30  code rebased.

**Comment by eldrin30** [comment]
As discussed, since its a new endpoint, kindly add the same in component.adoc file

**Comment by muhammadali9699** [comment]
@eldrin30 Documentation added please verify.

**Comment by eldrin30** [comment]
Code looks good :)

**Pull request merged**